tests/test_events.py

```diff
@@ -87,6 +87,7 @@ def test_event_with_duration():
         )
     )
 
-    assert 'DURATION:PT30M' in event
-    assert 'DTEND' not in event
-    assert 'DTSTART' in event
+    assert 'DURATION:PT30M' in event.serialize()
```

Comment:
Maybe serialize the event once in the beginning of the test, and do subsequent checks against that.